### ISSUE TITLE
Easier cURL uploads

### ISSUE THREAD

**Issue by BHSPitMonkey** [DHowett/ghostbin#30]
It would be fantastic if a simple cURL one-liner like this worked:

```bash
curl 'https://ghostbin.com/paste/new' -F 'text=@main.c'
```

That would be possible if that endpoint could accept a `multipart-form-data` style request.

Also, the above command would still require some additional flags/parsing to grab the `Location` header and build a usable URL. It would be great if the response body contained the fully-qualified paste URL, so that the above `curl` command would simply spit out the URL to stdout. A more fleshed-out example of what I'm hoping for:

```
$ curl 'https://ghostbin.com/paste/new' -F 'lang=C' -F 'text=@main.c'
https://ghostbin.com/paste/abcde
$
```

**Comment by DHowett** [comment]
ghost.sh uses cURL; it shouldn't be too difficult to extract an appropriate
command line from that. Apart from getting Location, though, it really is
fairly straightforward like your example.

**Comment by BHSPitMonkey** [comment]
Well the hope was that this can be done painlessly without needing to ask someone to fetch and run a helper script -- and d'oh, I forgot that `--data-urlencode` would obviate the need to change the Content-Type. I guess the only wishlist item here would be making the fully-qualified URL easier to get with fewer keystrokes (still haven't found a likeable one-liner for this).

**Comment by DHowett** [comment]
I didn't mean to imply that one might need to get the script ;) watch this
space, I'll document a bit more when I get home.
In the interim, passing -vv will show the response headers, including
location.

**Comment by BHSPitMonkey** [comment]
Yep, I know I can inspect the `Location` header with `-v`, but then you have to mangle that into something useful either by hand or by chaining some commands to do the parsing for you, and it ceases to be a fingers-on-keyboard friendly exercise and you may as well be using a helper script or relying on copypasta. It's superficial, but it's kind of a DX thing to me.
